app/controllers/students_controller.rb

```diff
@@ -0,0 +1,66 @@
+class StudentsController < ApplicationController
+	before_action :authenticate_teacher!,except: [:welcome]
+	before_action :authenticate_student!,only: [:welcome]
+	before_action :set_student,except: [:welcome,:list,:new,:create]
+	
+  def edit
+	end
+	
+  def show
+		
+	end
+	
+  def new
+		@student=Student.new
+	end
+
+	def create
+   @student=Student.new(student_params)
+
+ 
+   if @student.save
+   
+      no=Time.now.year.to_s+@student.standard_id.to_s+("0".*(4-@student.id.to_s.length))+@student.id.to_s
```

Comment:
What is variable 'no', how can i know purpose of this variable.
variable name should be meaningful 
set appropriate variable name.